src/unix/mediactrl.cpp

```diff
@@ -301,19 +311,32 @@ extern "C" {
 static gint gtk_window_realize_callback(GtkWidget* widget,
                                         wxGStreamerMediaBackend* be)
 {
+    window_id_type window_id;
+
     gdk_flush();
 
     GdkWindow* window = gtk_widget_get_window(widget);
     wxASSERT(window);
 
+#ifdef GDK_WINDOWING_X11
+#ifdef __WXGTK3__
+    if ( GDK_IS_X11_WINDOW(window) )
+#endif
+    {
+        window_id = (window_id_type)GDK_WINDOW_XID(window);
+    }
+#endif
+#ifdef GDK_WINDOWING_WAYLAND
+    if ( GDK_IS_WAYLAND_WINDOW(window) )
+    {
+        window_id = (window_id_type)gdk_wayland_window_get_wl_surface(window);
+    }
+#endif
+
```

Comment:
This looks pretty good to me.  My only suggestion would be to perhaps move this logic for getting the window_id to a separate function and then use that function in both places where you need the window_id.

Reply:
I integrated your suggestion in a function named get_gdk_window_id_from_widget.
Please let me know if you have further suggestions.

Reply:
actually, it looks that similar changes shoudl be applyed to mediactrl_gstplayer.cpp, is it correct?

Reply:
Your changes look good to me, nice work!  Yes, technically mediactrl_gstplayer.cpp needs a similar change for Wayland support.  Thus, it would be nice to share that new function you wrote between both files.  Not sure the best way to do that - it's probably not worth creating a new file for it, but perhaps there's a good place in some common file where it could live as  it may be useful in other places too.  @vadz or @paulcor might have a suggestion.

Reply:
In file mediactrl.cpp, the code use gst_video_overlay_set_window_handle ( => guintptr window_id) for gstreamer 1.x and gst_x_overlay_set_xwindow_id (gulong xwindow_id) for gstremer 0.10.x.
In file mediactrl.cpp, the code use gst_player_video_overlay_video_renderer_set_window_handle and gst_player_video_overlay_video_renderer_new (both using gpointer xwindow_id).
I reworked the code to use exactly the same function get_gdk_window_id_from_widget in both the files: this should make easy export it externally.
Looking forward for further suggestions...

Reply:
Yes, it would be definitely better to avoid duplicating this code. The simplest way to do it would be to create a new header under `include/wx/gtk/private` and put it there and make it inline. Please also call it something like `wxGtkGetIdFromWidget()`, i.e. use wx naming convention rather than GTK one to make it clear that it's our function and not anything from GTK.

TIA!

Reply:
@varigigi Just to avoid doing redundant work, do you plan to do/are in progress of doing this refactoring already by chance?

Reply:
@vadz, sorry for my absence, it's a crazy week at work.
If not too late, I should be able propose a new patch over the weekend.

Reply:
Sure, TIA!

Reply:
@vadz, beyond integrating your suggestions, I've partially reworked the function SetupXOverlay in file src/unix/mediactrl.cpp, trying to make it more readable.
Looking forward for further suggestions.